src/test/java/com/yelp/nrtsearch/server/grpc/LuceneServerTest.java

```diff
@@ -854,6 +856,56 @@ public void testSearchBasic() throws IOException, InterruptedException {
     checkHits(secondHit);
   }
 
+  @Test
+  public void testSearchFetchingAllFieldsWithWildcard() throws IOException, InterruptedException {
+    GrpcServer.TestServer testAddDocs =
+        new GrpcServer.TestServer(grpcServer, true, Mode.STANDALONE);
+    // 2 docs addDocuments
+    testAddDocs.addDocuments();
+    // manual refresh
+    grpcServer
+        .getBlockingStub()
+        .refresh(RefreshRequest.newBuilder().setIndexName(grpcServer.getTestIndex()).build());
+
+    // support format of ["*"]
+    SearchResponse searchResponse =
+        grpcServer
+            .getBlockingStub()
+            .search(
+                SearchRequest.newBuilder()
+                    .setIndexName(grpcServer.getTestIndex())
+                    .setStartHit(0)
+                    .setTopHits(10)
+                    .addAllRetrieveFields(Arrays.asList("*"))
+                    .build());
+
+    assertEquals(2, searchResponse.getTotalHits().getValue());
+    assertEquals(2, searchResponse.getHitsList().size());
+    SearchResponse.Hit firstHit = searchResponse.getHits(0);
+    checkHits(firstHit);
+    SearchResponse.Hit secondHit = searchResponse.getHits(1);
+    checkHits(secondHit);
+
+    // support format of "*"
+    SearchResponse searchResponseWithWildcard =
+        grpcServer
+            .getBlockingStub()
+            .search(
+                SearchRequest.newBuilder()
+                    .setIndexName(grpcServer.getTestIndex())
+                    .setStartHit(0)
+                    .setTopHits(10)
+                    .addRetrieveFields("*")
+                    .build());
+
+    assertEquals(2, searchResponseWithWildcard.getTotalHits().getValue());
+    assertEquals(2, searchResponseWithWildcard.getHitsList().size());
+    SearchResponse.Hit firstHitPlainStar = searchResponseWithWildcard.getHits(0);
```

Comment:
nit: plainStar in the variable name does not seem very descriptive.

Reply:
Thanks, changed the var names to `firstHitWithArrOfWildcard` and `firstHitWithStrOfWildcard` respectively, how do they sound now?